pkg/controller/servicemesh/memberroll/prometheus.go

```diff
@@ -0,0 +1,146 @@
+package memberroll
+
+import (
+	"context"
+	"fmt"
+
+	pkgerrors "github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+
+	"github.com/maistra/istio-operator/pkg/controller/common"
+)
+
+const (
+	prometheusConfigMapName         = "prometheus"
+	prometheusConfigurationFilename = "prometheus.yml"
+
+	prometheusScrapeConfigKeyName   = "scrape_configs"
+	prometheusScrapeSDConfigKeyName = "kubernetes_sd_configs"
+)
+
+type PrometheusReconciler interface {
+	reconcilePrometheus(ctx context.Context, prometheusCMName, prometheusNamespace string, configuredMembers []string) error
+}
+
+type defaultPrometheusReconciler struct {
+	Client client.Client
+}
+
+func (r *defaultPrometheusReconciler) reconcilePrometheus(ctx context.Context, prometheusCMName, prometheusNamespace string, members []string) error {
+	reqLogger := common.LogFromContext(ctx)
+	reqLogger.Info("Attempting to get Prometheus ConfigMap", "prometheusNamespace", prometheusNamespace, "prometheusCMName", prometheusCMName)
+
+	cm := &corev1.ConfigMap{}
+
+	err := r.Client.Get(ctx, client.ObjectKey{Name: prometheusCMName, Namespace: prometheusNamespace}, cm)
+	if err != nil {
+		if meta.IsNoMatchError(err) || errors.IsNotFound(err) || errors.IsGone(err) {
+			reqLogger.Info("Prometheus ConfigMap '%s/%s' does not exist", prometheusNamespace, prometheusCMName)
+			return nil
+		}
+		return pkgerrors.Wrap(err, "error retrieving Prometheus ConfigMap from mesh")
+	}
+
+	data := make(map[string]interface{})
+
+	err = yaml.Unmarshal([]byte(cm.Data[prometheusConfigurationFilename]), &data)
+
+	if err != nil {
+		return pkgerrors.Wrap(err, "error unmarshaling prometheus.yml")
+	}
+
+	f, ok, err := unstructured.NestedFieldNoCopy(data, prometheusScrapeConfigKeyName)
+
+	var scrapes []interface{}
+
+	if err == nil && ok {
+		scrapes, ok = f.([]interface{})
+
+		if !ok {
+			return fmt.Errorf("error no scrape_configs found from %v", f)
+		}
+	} else {
+		return pkgerrors.Wrap(err, "error getting scrape_configs from ConfigMap")
+	}
+
+	scrapingMembers := append(members, prometheusNamespace)
+
```

Comment:
@jewertow, The `PrometheusNamespace` in here is the same as `meshNamespace`. We start the Prometheus pod in the same namespace as the control plane.